### PR TITLE
Costmap Filter enabling service

### PR DESCRIPTION
**Pull request by AlexeyMerzlyakov**
Adding costmap filters enabling/disabling service.
For the basis it was taken already existing "ros2/common_interfaces/std_srvs/srv/SetBool.srv":
```
bool data # e.g. for hardware enabling / disabling
---
bool success   # indicate successful run of triggered service
string message # informational, e.g. for error messages
```

---

## Basic Info

| Info | Please fill out this column |
| ------ | ----------- |
| Ticket(s) this addresses   | #3194 |
| Primary OS tested on | Ubuntu 20.04 with ROS2 Rolling built from sources onboard |
| Robotic platform tested on | TB3 simulation, unit testing, code coverage |

---

#### For Maintainers: 
- [ ] Check that any new parameters added are updated in navigation.ros.org
- [ ] Check that any significant change is added to the migration guide
- [ ] Check that any new features **OR** changes to existing behaviors are reflected in the tuning guide
- [ ] Check that any new functions have Doxygen added
- [ ] Check that any new features have test coverage
- [ ] Check that any new plugins is added to the plugins page
- [ ] If BT Node, Additionally: add to BT's XML index of nodes for groot, BT package's readme table, and BT library lists

### CODE REVIEW

File: nav2_costmap_2d/plugins/costmap_filters/costmap_filter.cpp [hunk 1]
```diff
@@ -63,15 +63,25 @@ void CostmapFilter::onInitialize()
...
     declareParameter("enabled", rclcpp::ParameterValue(true));
+    declareParameter("enable_service", rclcpp::ParameterValue("enable"));
     declareParameter("filter_info_topic", rclcpp::PARAMETER_STRING);
     declareParameter("transform_tolerance", rclcpp::ParameterValue(0.1));
 
     // Get parameters
     node->get_parameter(name_ + "." + "enabled", enabled_);
     filter_info_topic_ = node->get_parameter(name_ + "." + "filter_info_topic").as_string();
+    std::string enable_service_name;
+    node->get_parameter(name_ + "." + "enable_service", enable_service_name);
```
Comment (SteveMacenski) [inline nav2_costmap_2d/plugins/costmap_filters/costmap_filter.cpp:74]: Don't parameterize the name if its going to be put under the namespace of the costmap filter anyway. That way these are always standardized service names that people can rely on. 

How about `toggle_filter`?

**Review by SteveMacenski — CHANGES_REQUESTED** [review]

**Comment by SteveMacenski** [comment]
Make sure to open a migration guide PR as well mentioning this exists.

File: nav2_costmap_2d/plugins/costmap_filters/costmap_filter.cpp (hunk 1 above)
Comment (AlexeyMerzlyakov) [reply nav2_costmap_2d/plugins/costmap_filters/costmap_filter.cpp]: Agree with this name. Made an update in https://github.com/ros-planning/navigation2/pull/3229/commits/33aeff9ebe1cd4c3e46a0ac2cbb81b3b07eb4db4

**Comment by AlexeyMerzlyakov** [comment]
> Make sure to open a migration guide PR as well mentioning this exists.

Thanks for reminding. Done in https://github.com/ros-planning/navigation.ros.org/pull/350.

**Pull request merged**